### DISCUSSION TITLE
Some questions about public transit support (GTFS, GTFS-RT, GBFS)

### DISCUSSION THREAD

**Discussion by TTelkdotcom** [valhalla/valhalla#4056] [Q&A] [answered]
Hi, we're just getting started with Valhalla. We're currently looking to upgrade our project to a new routing system and we are considering Valhalla, but as public transit is essential for what we have in mind, here are some questions that cam up.

I saw that you were already asked late last year on a timeline for GTFS, but I wanted to ask wether you have an updated estimate to how long it will take to publish the new public transit API with GTFS? Do you also plan on supporting GTFS-RT or GBFS and if so how far is development and how long may it take for finalisation of those?

Thanks.

**Comment by nilsnolde** [comment, marked as answer]
>  but I wanted to ask wether you have an updated estimate to how long it will take to publish the new public transit API with GTFS?

It is "published", in the sense of being usable (and merged in `master`) with a few caveats. Those are lined out mainly here: #4003 and the linked issues & PRs. We're still looking to merge a few PRs, esp #4054 as that seems to enable hierarchy support again, which seems broken right now with multiple feeds (though I can't tell how multiple feeds would trigger that, but it came up during #4055 ..). 

Even when those PR(s) are merged, it'll still be beta. We're eventually looking to include a few (ever increasing number of) GTFS feeds at the public instance. It's hard to really catch bugs though when using it, at least time/departure-related bugs, so it'll need better unit testing. We're getting there.

> Do you also plan on supporting GTFS-RT or GBFS and if so how far is development and how long may it take for finalisation of those?

That'd definitely be interesting. But would require proper funding, it's not a low-effort ask. We're open for funding to work such features. If there's none, I'd focus on hardening static GTFS support, e.g. pulling out the transit builder from the general build pipeline to much quicker update the transit network without re-building the OSM graph. Also ideally getting rid of the PBF intermediate stage as that takes ages.
